src/Groups/GAPGroups.jl

```diff
@@ -756,123 +758,130 @@ function Base.rand(rng::Random.AbstractRNG, C::GroupConjClass{S,T}) where S wher
 end
 
 """
-    conjugacy_classes_subgroups(G::Group)
+    subgroup_classes(G::GAPGroup; order::T = ZZRingElem(-1)) where T <: IntegerUnion
 
-Return the vector of all conjugacy classes of subgroups of G.
+Return the vector of all conjugacy classes of subgroups of G or,
+if `order` is positive, the classes of subgroups of this order.
 
 # Examples
 ```jldoctest
 julia> G = symmetric_group(3)
 Sym(3)
 
-julia> conjugacy_classes_subgroups(G)
+julia> subgroup_classes(G)
 4-element Vector{GAPGroupConjClass{PermGroup, PermGroup}}:
  Conjugacy class of permutation group in G
  Conjugacy class of permutation group in G
  Conjugacy class of permutation group in G
  Conjugacy class of permutation group in G
+
+julia> subgroup_classes(G, order = ZZRingElem(2))
+1-element Vector{GAPGroupConjClass{PermGroup, PermGroup}}:
+ Conjugacy class of permutation group in G
 ```
 """
-function conjugacy_classes_subgroups(G::GAPGroup)
+function subgroup_classes(G::GAPGroup; order::T = ZZRingElem(-1)) where T <: IntegerUnion
   L = Vector{GapObj}(GAPWrap.ConjugacyClassesSubgroups(G.X))
+  if order != -1
+    L = [x for x in L if GAPWrap.Order(GAPWrap.Representative(x)) == order]
```

Comment:
Does the `==` comparison work when `order` is a `ZZRingElem` and the group is a `GapObj` (and not an `Int`)? I would have thought not (it might be hard to add a test case for that, though... perhaps if `G` is a huge group of prime order?)

Reply:
Right, this comparison is not safe, the `GAP.Obj` on the left side should better be converted to the type `T`.
Concerning  a test case: In principle this would work, but there is a bug on the GAP side, see gap-system/gap/issues/5638.